### PR TITLE
Collected small changes and fixes

### LINKED ISSUE #3476: [BUG] thermo_modify format accepts too many fields
**Summary**

The `thermo_modify` command currently allows the user to specify a format string for one too many fields. Example is below.

**LAMMPS Version and Platform**

Latest develop branch (synchronized this afternoon, 10/6/2022).

**Expected Behavior**

The following input script should produce an error:
```
region simbox block 0 5 0 5 0 5
create_box 1 simbox
thermo_style custom step etotal
thermo_modify format 3 "%G"
```
The last line should try to access the third element of a two-element vector, producing an error.

**Actual Behavior**
No error message.

**Further Information, Files, and Links**

I think I know how to fix this (it appears to be a simple off-by-one error in the bounds-checking), but just in case someone says, "NO, DON'T FIX IT THAT WAY!" I am posting this here first. I'm planning to change the error-checking bounds, not resize `format_column_user`. I see in other parts of thermo_modify.cpp that there is an "invisible" field containing the volume of the box, which is why this doesn't cause a more serious problem.

I am also writing a feature (and tripped this bug while testing it) that will allow things like `thermo_modify format 3*5 "%G"` to be used to set the format of consecutive columns at the same time.

### LINKED ISSUE #3484: [BUG] CMake issue with clang-format on ubuntu 22.04
It seems that CMake is not able to identify the correct clang-format version if the output of `clang-format --version` is something like `Ubuntu clang-format version 14.0.0-1ubuntu1` using 
```
    string(REGEX REPLACE "clang-format version ([0-9.]+).*"
                         "\\1"
                         ClangFormat_VERSION
                         "${clang_format_version}")
```
in `cmake/Modules/FindClangFormat.cmake`. I only get the message `Could NOT find ClangFormat: Found unsuitable version "0.0", but required is at least "8.0" (found /usr/bin/clang-format)`. Unfortunately, I don't know too much of regex nor CMake to modify it accordingly. Maybe this is also an issue with other libraries and packages?

### LINKED ISSUE #3496: [BUG] Hyperlinks to functions in documentation aren't resolving the first time
**Summary**

When I build the documentation, the links to Fortran functions in doc/html/Fortran.html don't work (e.g., the links to `file()`, `command()`, `commands_list()`, and `commands_string()` under the "1.3.3. Executing LAMMPS commands" heading). Neither do the ones in doc/fasthtml/Fortran.html when I do a `make fasthtml`.

However, they *do* work if I issue the following commands (from the "doc" directory):
```
make clean
make fasthtml
touch src/Fortran.rst
make fasthtml
```
The HTML files after the first `make fasthtml` do not have hyperlinked cross-references, but those after the second one do. I suspect there is an operation (or two) that either needs to be done twice, or in another order.

**LAMMPS Version and Platform**

LAMMPS (23 Jun 2022) [also happens with latest "develop" branch on October 21, 2022]

I'm running this on Fedora 36.

**Expected Behavior**

Hyperlinks work after first run and in the `make html` version.

**Actual Behavior**

Formatting is fine, but hyperlinks are not created in the `make fasthtml` unless the file is re-touched and `make fasthtml` is run again. Hyperlinks never work in `make html`, from the tests I have done.

**Steps to Reproduce**

 (1) cd [lammps_root]/doc
 (2) `make clean`
 (3) `make html fasthtml`
 (4) Direct browser to [lammps_root]/doc/html/Fortran.html
 (5) Scroll down to "Executing LAMMPS commands" and hover over `*command()*` or `*file()*` (it won't be a link)
 (6) Direct browser to [lammps_root]/doc/fasthtml/Fortran.html
 (7) Scroll down to "Executing LAMMPS commands" and hover over `*command()*` or `*file()*` (it won't be a link)
 (8) `touch [lammps_root]/doc/src/Fortran.rst`
 (9) `make fasthtml html`
 (10) Repeat steps (4) through (7) or reload the pages from before; it will look the same, except this time step (7) will give you hyperlinks that work

### LINKED ISSUE #3499: VTK deprecation notice is printed
Configure log has this:
```
CMake Deprecation Warning at /usr/local/lib/cmake/vtk-9.2/vtk-use-file-deprecated.cmake:1 (message):
  The `VTK_USE_FILE` is no longer used starting with 8.90.
Call Stack (most recent call first):
  Modules/Packages/VTK.cmake:2 (include)
  CMakeLists.txt:503 (include)
```

VTK-9.2.2
FreeBSD 13.1

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This pull request combines multiple small changes and fixes.

**Related Issue(s)**

Fixes #3499 
Fixes #3496 
Fixes #3484 
Fixes #3476 
Supersedes and thus closes #3483 
Supersedes and thus closes #3493 

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

Some regression tests will fail because of the different reneighboring steps causing differences in forces and thus diverging trajectories.

**Implementation Notes**

The following individual changes are inlcuded:
- change the default setting for neighbor lists to reduce delay=10 to delay=0, add note to documentation
- delete *atomfile* style variables with the `clear` command to avoid segmentation faults on subsequent `read_data` calls
- multiple small fixes for issues related to PR #3478 
- correctly detect the `clang-format` version on Ubuntu 22.04LTS with CMake (see issue #3484)
- update list and CMake script code for installing include files from the main source folder with "make install"
- update PLUMED package library to version 2.8.1 and also support 2.7.5
- small FreeBSD portability enhancement
- report the new communication cutoff in the warning when it is extended for some of the GPU pair styles
- increase the communication cutoff for TIP4P pair styles to avoid errors (adapted from PR #3483)
- corrects column selection limits in thermo_modify format (see issue #3476)
- provide access to `lammps_config_has_package()` in the python module
- stop with an error when trying to use `-D ENABLE_TESTING=on` with too old compilers (e.g. GNU 4.8.x)
- skip `FortranOpen` tests if no suitable F90 MPI module is found.
- skip some unit tests that require the MOLECULE package in case it is not available
- improve a few error messages
- more detailed message with the replicate command (alternative to PR #3493)
- reformat sources with clang-format
- implement workaround resulting in proper hyperlinks to Fortran functions in the manual (addresses issue #3496)

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

### CODE REVIEW

**Comment by evoyiatzis** [comment]
In doc/src/run_style.rst line 104 there is a spelling error "its onn optimal layout" instead of "its own optimal layout"

**Comment by akohlmey** [comment]
> In doc/src/run_style.rst line 104 there is a spelling error "its onn optimal layout" instead of "its own optimal layout"

Thanks. I found that this passed the spell check because of an incorrect entry in the false positives list.

File: doc/src/neigh_modify.rst [hunk 1]
```diff
@@ -71,30 +71,52 @@ Description
...
+since the last neighbor list build.
+
+.. admonition:: Impact of neighbor list settings
+   :class: note
+
+   The choice of neighbor list settings can have a significant impact on
+   the (parallel) performance of LAMMPS and the correctness of the
+   simulation results.  Since building the neighbor lists is time
+   consuming, doing it less frequently can speed up a calculation.  If
+   the lists are rebuilt too infrequent, however, interacting pairs are
```
Comment (sjplimp) [inline doc/src/neigh_modify.rst]: infrequently

File: doc/src/neigh_modify.rst [hunk 2]
```diff
@@ -71,30 +71,52 @@ Description
...
+   missed and thus the resulting pairwise interactions incorrect.  The
```
Comment (sjplimp) [inline doc/src/neigh_modify.rst]: interacting pairs can potentially be missed

File: doc/src/neigh_modify.rst [hunk 3]
```diff
@@ -71,30 +71,52 @@ Description
...
+   optimal settings depend on many factors like the properties of the
+   simulated system (density, geometry, topology, temperature,
+   pressure), the force field parameters and settings, the size of the
+   timestep, neighbor list skin distance and more.  The default settings
+   are chosen to be very conservative to promote correctness of the
```
Comment (sjplimp) [inline doc/src/neigh_modify.rst]: to guarantee correctness

File: doc/src/neigh_modify.rst [hunk 4]
```diff
@@ -71,30 +71,52 @@ Description
...
+   simulation.  They depend on the *check* flag heuristics to reduce the
+   number of neighbor list rebuilds at a minor expense for executing the
+   check.  Determining the correctness of a specific coice of neighbor
```
Comment (sjplimp) [inline doc/src/neigh_modify.rst]: choice

**Review by sjplimp — CHANGES_REQUESTED** [review]
a few typos & rewordings

**Comment by akohlmey** [comment]
> a few typos & rewordings

changes are applied

File: src/replicate.cpp [hunk 5]
```diff
@@ -47,15 +47,16 @@ void Replicate::command(int narg, char **arg)
...
-
   // nrep = total # of replications
 
   int nx = utils::inumeric(FLERR,arg[0],false,lmp);
   int ny = utils::inumeric(FLERR,arg[1],false,lmp);
   int nz = utils::inumeric(FLERR,arg[2],false,lmp);
   int nrep = nx*ny*nz;
 
+  if (me == 0)
+    utils::logmesg(lmp,"Replicating atoms for a {}x{}x{} = {} times size system...\n", nx, ny, nz, nrep);
```
Comment (sjplimp) [inline src/replicate.cpp]: Replication is creating a {}x{}x{} = {}x larger system

**Review by sjplimp — CHANGES_REQUESTED** [review]
what did the TIP4P cutoff become ?

File: src/replicate.cpp (hunk 5 above)
Comment (akohlmey) [reply src/replicate.cpp]: done

**Comment by akohlmey** [comment]
> what did the TIP4P cutoff become ?

Same as for the TIP4P GPU style: pair_style cutoff + q-distance + OH-bond distance + neighbor skin

File: src/replicate.cpp [hunk 6]
```diff
@@ -47,15 +47,16 @@ void Replicate::command(int narg, char **arg)
...
-
   // nrep = total # of replications
 
   int nx = utils::inumeric(FLERR,arg[0],false,lmp);
   int ny = utils::inumeric(FLERR,arg[1],false,lmp);
   int nz = utils::inumeric(FLERR,arg[2],false,lmp);
   int nrep = nx*ny*nz;
 
+  if (me == 0)
+    utils::logmesg(lmp,"Replication is creating a {}x{}x{} = {}x larger system...\n",nx,ny,nz,nrep);
```
Comment (sjplimp) [inline src/replicate.cpp]: sorry, though better of {}x larger, so instead:
= {} times larger system

Comment (akohlmey) [reply src/replicate.cpp]: done as well

**Review by sjplimp — APPROVED** [review]

**Review by stanmoore1 — APPROVED** [review]

**Pull request merged**